syntax = "proto3";

package messageservice;

service MessageServer{
  rpc GetContect(MsgRequest) returns (MsgReply) {}

  rpc DifferTest(log) returns (result) {}

  rpc MapKeyCompare(log) returns (result) {}
}

message MsgRequest{
  string request = 1;
}

message MsgReply{
  string reply = 1;
}

message log {
  string base = 1;
  string test = 2;
  string name_of_message_descriptor = 3;
  repeated bytes file_descriptor_proto = 4;

  // For ignoreCriteria the filed_list[0] is binary option:
  // WHITE OR BLACK represents white-list and black-list or black-list
  // the following string is the field name that used in hte ignore criteria.
  repeated string ignore_field_list = 5;

  // for regex match criteria
  string regex = 6;

  // for Map Key compare
  repeated string field_names = 7;
}

message result {
  string res = 1;
}



